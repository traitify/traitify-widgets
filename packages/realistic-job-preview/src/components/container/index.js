import PropTypes from "prop-types";
import Context from "lib/context";
import ErrorBoundary from "./error-boundary";
import Theme from "./theme";
import useReducer from "./use-reducer";

function Container({children, className = null, ...props}) {
  const state = useReducer({...props});

  return (
    <Context.Provider value={state}>
      <Theme className={className}>
        <ErrorBoundary>{children}</ErrorBoundary>
      </Theme>
    </Context.Provider>
  );
}

Container.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string
};

export default Container;
