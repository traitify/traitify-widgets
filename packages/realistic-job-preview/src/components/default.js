import useAssessment from "lib/hooks/use-assessment";
import Results from "./results";
import Survey from "./survey";

export default function Default() {
  const assessment = useAssessment();

  if(!assessment) { return null; }
  if(assessment.completedAt) { return <Results />; }

  return <Survey />;
}
