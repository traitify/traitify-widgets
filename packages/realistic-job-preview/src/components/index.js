import Container from "./container";
import Default from "./default";

export default {
  Container,
  Default
};
