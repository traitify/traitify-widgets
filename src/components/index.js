import Container from "./container";
import Default from "./default";
import PersonalityArchetypeHeading from "./personality/archetype/heading";
import PersonalityArchetypeSkills from "./personality/archetype/skills";
import PersonalityArchetypeTips from "./personality/archetype/tips";
import PersonalityBaseDetails from "./personality/base/details";
import PersonalityBaseHeading from "./personality/base/heading";
import PersonalityCareerContainer from "./personality/career/container";
import PersonalityCareerDetails from "./personality/career/details";
import PersonalityCareerFilter from "./personality/career/filter";
import PersonalityCareerList from "./personality/career/list";
import PersonalityCareerModal from "./personality/career/modal";
import PersonalityDimensionChart from "./personality/dimension/chart";
import PersonalityDimensionDetails from "./personality/dimension/details";
import PersonalityDimensionList from "./personality/dimension/list";
import PersonalityRecommendationChart from "./personality/recommendation/chart";
import PersonalityTraitDetails from "./personality/trait/details";
import PersonalityTraitList from "./personality/trait/list";
import PersonalityTypeChart from "./personality/type/chart";
import PersonalityTypeList from "./personality/type/list";
import AttractReport from "./report/attract";
import CandidateReport from "./report/candidate";
import EmployeeReport from "./report/employee";
import ManagerReport from "./report/manager";

export default {
  Container,
  Default,
  Personality: {
    Archetype: {
      Heading: PersonalityArchetypeHeading,
      Skills: PersonalityArchetypeSkills,
      Tips: PersonalityArchetypeTips
    },
    Base: {
      Details: PersonalityBaseDetails,
      Heading: PersonalityBaseHeading
    },
    Career: {
      Container: PersonalityCareerContainer,
      Details: PersonalityCareerDetails,
      Filter: PersonalityCareerFilter,
      List: PersonalityCareerList,
      Modal: PersonalityCareerModal
    },
    Dimension: {
      Chart: PersonalityDimensionChart,
      Details: PersonalityDimensionDetails,
      List: PersonalityDimensionList
    },
    Recommendation: {
      Chart: PersonalityRecommendationChart
    },
    Trait: {
      Details: PersonalityTraitDetails,
      List: PersonalityTraitList
    },
    Type: {
      Chart: PersonalityTypeChart,
      List: PersonalityTypeList
    }
  },
  Report: {
    Attract: AttractReport,
    Candidate: CandidateReport,
    Employee: EmployeeReport,
    Manager: ManagerReport
  }
};
