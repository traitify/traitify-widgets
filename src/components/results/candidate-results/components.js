export Guide from "components/results/guide";
export PersonalityArchetype from "components/results/personality/archetype/heading";
export PersonalityTips from "components/results/personality/archetype/tips";
export PersonalityDimensionChart from "components/results/personality/dimension/chart";
export PersonalityDimensionColumn from "components/results/personality/dimension/column";
export PersonalityDimensionColumns from "components/results/personality/dimension/chart";
export PersonalityDimensionDetails from "components/results/personality/dimension/details";
export PersonalityDimensions from "components/results/personality/dimension/list";
export PersonalityRecommendationChart from "components/results/personality/recommendation/chart";
export PersonalityTraits from "components/results/personality/trait/list";
export CandidateResults from "./index";
