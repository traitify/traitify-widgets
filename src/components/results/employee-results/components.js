export PersonalityArchetype from "components/results/personality/archetype/heading";
export PersonalitySkills from "components/results/personality/archetype/skills";
export PersonalityTips from "components/results/personality/archetype/tips";
export PersonalityDimensionChart from "components/results/personality/dimension/chart";
export PersonalityDimensionColumn from "components/results/personality/dimension/column";
export PersonalityDimensionColumns from "components/results/personality/dimension/chart";
export PersonalityDimensionDetails from "components/results/personality/dimension/details";
export PersonalityDimensions from "components/results/personality/dimension/list";
export EmployeeResults from "./index";
