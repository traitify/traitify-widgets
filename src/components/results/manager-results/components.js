export Guide from "components/results/guide";
export PersonalityRecommendationChart from "components/results/personality/recommendation/chart";
export ManagerResults from "./index";
