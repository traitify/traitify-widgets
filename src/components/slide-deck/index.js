import PropTypes from "prop-types";
import withTraitify from "lib/with-traitify";
import Cognitive from "./cognitive";
import Personality from "./personality";

function SlideDeck(props) {
  if(props.getOption("surveyType") === "cognitive") { return <Cognitive {...props} />; }

  return <Personality {...props} />;
}

SlideDeck.propTypes = {getOption: PropTypes.func.isRequired};

export {SlideDeck as Component};
export default withTraitify(SlideDeck);
