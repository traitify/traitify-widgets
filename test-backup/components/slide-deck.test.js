import {Component} from "components/slide-deck";
import ComponentHandler from "support/component-handler";

jest.mock("components/slide-deck/cognitive", () => (() => (<div className="mock">Cognitive</div>)));
jest.mock("components/slide-deck/personality", () => (() => (<div className="mock">Personality</div>)));
jest.mock("lib/with-traitify");

describe("SlideDeck", () => {
  let props;

  beforeEach(() => {
    props = {
      getOption: jest.fn().mockName("getOption"),
      ui: {
        current: {},
        off: jest.fn().mockName("off"),
        on: jest.fn().mockName("on"),
        trigger: jest.fn().mockName("trigger")
      }
    };
  });

  it("renders cognitive", () => {
    props.getOption.mockReturnValue("cognitive");
    const component = new ComponentHandler(<Component {...props} />);

    expect(component.tree).toMatchSnapshot();
  });

  it("renders personality", () => {
    const component = new ComponentHandler(<Component {...props} />);

    expect(component.tree).toMatchSnapshot();
  });
});
