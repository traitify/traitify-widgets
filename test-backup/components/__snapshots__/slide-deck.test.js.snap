// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SlideDeck renders cognitive 1`] = `
<div
  className="mock"
>
  Cognitive
</div>
`;

exports[`SlideDeck renders personality 1`] = `
<div
  className="mock"
>
  Personality
</div>
`;
